rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java

```diff
@@ -259,6 +273,25 @@ public boolean restore_state(InputObjectState os, int ot) {
                 }
             }
 
+            /*
+             * If the time limit has already been reached then the difference between now and the scheduled
+             * abort time will be negative. Since scheduling a task with a negative time will run it immediately
+             * we must ensure that the setTimeLimit call is placed after the state has been fully re-hydrated.
+             */
+            if (finishTime != null) {
+                long ttl = ChronoUnit.MILLIS.between(LocalDateTime.now(ZoneOffset.UTC), finishTime);
+
+                if (ttl <= 0) {
+                    System.out.printf("TIMER has expired since last reload%n");
```

Comment:
Should it be better to log these statements instead of system.out?

Reply:
Good catch, this should be using the logger framework

Reply:
+1 (and to the other occurrences of printf in this PR)